packages/manager/apps/dedicated/client/app/ip/components/list/firewall/ip-ip-firewall.html

```diff
@@ -75,23 +108,196 @@ <h4 data-translate="ip_firewall_title"></h4>
                 </tr>
             </thead>
 
-            <tbody data-ng-if="rulesLoading">
+            <tbody data-ng-if="IpFirewallCtrl.rulesLoading">
                 <tr>
                     <td class="text-center" colspan="9">
                         <oui-spinner data-size="s"></oui-spinner>
                     </td>
                 </tr>
             </tbody>
 
-            <tbody data-ng-if="rules.count == 0 && !rulesLoading">
-                <tr>
-                    <td class="text-center" colspan="9">
+            <tbody
+                data-ng-if="!IpFirewallCtrl.rulesLoading && IpFirewallCtrl.rules.count >= 0"
+            >
+                <tr data-ng-if="IpFirewallCtrl.isAddingRule">
+                    <td>
+                        <select
+                            class="form-control"
+                            id="sequenceSelect"
+                            name="sequenceSelect"
+                            data-ng-model="IpFirewallCtrl.rule.sequence"
+                            data-ng-options="value as value.name for value in (IpFirewallCtrl.constants.sequences | orderBy:'name')"
+                            data-ng-change="IpFirewallCtrl.isFirewallRuleFormValid()"
+                        ></select>
+                    </td>
+                    <td>
+                        <select
+                            class="form-control"
+                            id="actionSelect"
+                            name="actionSelect"
+                            data-ng-model="IpFirewallCtrl.rule.action"
+                            data-ng-options="action as (('ip_firewall_rule_' + action) | translate) for action in IpFirewallCtrl.constants.actions"
+                            data-ng-change="IpFirewallCtrl.isFirewallRuleFormValid()"
+                        ></select>
+                    </td>
+                    <td>
+                        <select
+                            id="protocolSelect"
+                            name="protocolSelect"
+                            class="form-control"
+                            data-ng-model="IpFirewallCtrl.rule.protocol"
+                            data-ng-change="IpFirewallCtrl.resetOptionField()"
+                            data-ng-options="value as ('ip_firewall_rule_protocol_' + value | translate) for value in IpFirewallCtrl.constants.protocols"
+                            required
+                        >
+                        </select>
```

Comment:
**nitpick:** Can you add `oui-select` and `oui-select__input` classes to maintain the uniform UX.
```suggestion
                    <td class="oui-select">
                        <select
                            class="form-control oui-select__input"
                            id="sequenceSelect"
                            name="sequenceSelect"
                            data-ng-model="IpFirewallCtrl.rule.sequence"
                            data-ng-options="value as value.name for value in (IpFirewallCtrl.constants.sequences | orderBy:'name')"
                            data-ng-change="IpFirewallCtrl.isFirewallRuleFormValid()"
                        ></select>
                    </td>
                    <td class="oui-select">
                        <select
                            class="form-control oui-select__input"
                            id="actionSelect"
                            name="actionSelect"
                            data-ng-model="IpFirewallCtrl.rule.action"
                            data-ng-options="action as (('ip_firewall_rule_' + action) | translate) for action in IpFirewallCtrl.constants.actions"
                            data-ng-change="IpFirewallCtrl.isFirewallRuleFormValid()"
                        ></select>
                    </td>
                    <td class="oui-select">
                        <select
                            id="protocolSelect"
                            name="protocolSelect"
                            class="form-control oui-select__input"
                            data-ng-model="IpFirewallCtrl.rule.protocol"
                            data-ng-change="IpFirewallCtrl.resetOptionField()"
                            data-ng-options="value as ('ip_firewall_rule_protocol_' + value | translate) for value in IpFirewallCtrl.constants.protocols"
                            required
                        >
                        </select>
```